synapse/lib/storm.py

```diff
@@ -153,3 +159,87 @@ class SudoCmd(Cmd):
     def runStormCmd(self, snap, genr):
         snap.elevated = True
         yield from genr
+
+class TaskCmd(Cmd):
+    '''
+    Queue a node for consumption by a persistent task queue.
+
+    Tasking nodes to a single service:
+
+        ask < some lift > | task fooService
+
+    Additional keyword args may be provided, as pairs, to the task:
+
+        ask < some lift > | task -k arg1 valu1 -k arg2 valu2 fooService
+
+    It is possible to provide multiple named tasks for a single node at once:
+
+        ask < some lift > | task fooService barService
+
+    The packaged data which is sent to the queue will contain the following data:
+    - "node": The packed version of the Node. This is None there are no prior
+    nodes in the Storm pipeline.
+    - "tid": A task identifier. This is a randomly generated guid.  All events
+    created from a single instance of this will have the same task identifier.
+    - "tick": The system time when the task was fired.
+    - "user": The user that fired the task. It may be None if authentication
+    is not enabled on the Cortex.
+    - "kwargs": A dictionary of additional keyword arguments provided by the
+    user at runtime. These are left as strings for a downstream consumer to
+    use.
+
+    This command will not function if the Cortex is not configured for a task
+    queuing endpoint.  The Cortex is only responsible for shipping the data to
+    the queue. It is not responsible for managing any downstream consumers of
+    that queue.
+    '''
+
+    name = 'task'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('name', nargs='*',
+                          help='Name of the task to fire.')
+        pars.add_argument('-k', '--kwarg', action='append', default=[], nargs=2,
+                          help='Additional keyword & value string arguments to include in the task message.')
+        return pars
+
+    def runStormCmd(self, snap, genr):
+        # Is tasking even enabled on this Cortex?
```

Comment:
reaching into `snap.core.tqueue` feels odd but we don't have any other way to do this type of check.